docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -0,0 +1,391 @@
+# Defining placement rules in SQL
+
+- Author(s):     [djshow832](https://github.com/djshow832) (Ming Zhang)
+- Last updated:  2020-06-24
+- Discussion at: https://docs.google.com/document/d/18Kdhi90dv33muF9k_VAIccNLeGf-DdQyUc8JlWF9Gok
+
+## Movitation
+
+TiDB supports placement rules, which can define the placement of data in a more flexible and more granular way. But it only provides configuration files to define them, and it’s complicated.
+
+This article proposes an approach to configure placement rules through statements. TiDB server parses the statements and notify PD to perform the change. In this way, usability can be improved.
+
+## Define placement rules
+
+There are 2 kinds of operations on the placement:
+
+* ADD: Add new `count` replicas to the table according to `label` and `role` configuration.
+* ALTER: Override the current replica configuration. Old configuration is discarded.
+
+They’re both achieved by executing `ALTER TABLE` statements.
+
+### Adding replicas
+
+Adding new replicas can be done by one or more `ADD PLACEMENT` clauses:
+
+```sql
+ALTER TABLE table_name
+	ADD PLACEMENT LABEL=label ROLE=role COUNT=count,
+	...
+```
+
+This statement indicates TiDB to add replicas for all data, including indexes, of table `table_name`.
+
+`ADD PLACEMENT` is just a part of alter options, just like `ADD COLUMN` or `ADD CONSTRAINT`.
+
+To define multiple roles at once, Multiple `ADD PLACEMENT` clauses can appear in a single `ALTER TABLE` statement, each for one Raft role:
+
+```sql
+ALTER TABLE table_name
+	ADD PLACEMENT LABEL="+zone=sh" ROLE=leader COUNT=1,
+	ADD PLACEMENT LABEL="-zone=sh" ROLE=follower COUNT=2;
+```
+
+This statement indicates PD to schedule the leader to `sh`, and do not schedule followers to `sh`.
+
+`ADD PLACEMENT` can be used to separate data on different data centers, which may help to reduce multi-region latency.
+
+Adding placement can be done through adding a placement rule in PD. The statement must wait until the PD returns a message.
```

Comment:
What's the behavior on TiDB if users enter `ctrl+c` in MySQL client?

Reply:
Just like other DDLs, it won't break this session. Only `CANCEL DDL JOB` will break it.
I'm assuming you're consider the disaster recovery. The DDL job is put into the job queue, so even it's interrupted, it will recover automatically by another DDL owner.